health-services/libraries/health-services-models/src/main/java/org/egov/common/models/project/TaskAction.java

```diff
@@ -0,0 +1,33 @@
+package org.egov.common.models.project;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public enum TaskAction {
+    CLOSED_HOUSEHOLD("CLOSED_HOUSEHOLD"),
+    LOCATION_CAPTURE("LOCATION_CAPTURE"),
+    OTHER("OTHER");
+
+    private String value;
+
+    TaskAction(String value) {
+        this.value = value;
+    }
+
+    @Override
+    @JsonValue
+    public String toString() {
+        return String.valueOf(value);
+    }
+
+    @JsonCreator
+    public static TaskAction fromValue(String text) {
```

Comment:
if you return null then if UI send wrong value you will never catch it.

Reply:
JsonCreator annotation is being used here, where the type is used in a property, it has @NotNull annotation, so it will throw error even when any string is given from request which does not have a enum value mapped to it.